draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,76 +60,333 @@ collector). Thus it is up to each deployment of DAP to securely implement its
 
 ~~~
-TODO(wangshan) Specify the structure of the extension.
+struct {
+    /* Info specific for a task. */
+    opaque task_info<1..2^8-1>;
+
+    /* A list of URLs relative to which an aggregator's API endpoints
+     can be found. Defined in I-D.draft-ietf-ppm-dap-02. */
+    Url aggregator_endpoints<1..2^16-1>;
+
+    /* This determines the query type for batch selection and the
+    properties that all batches for this task must have. Defined in
+     I-D.draft-ietf-ppm-dap-02. */
+    QueryConfig query_config;
+
+    /* The maximum number of times a batch of reports may be queried by
+    the Collector. */
+    uint16 max_batch_lifetime;
+
+    /* Time up to which Clients are allowed to upload to this task. See
+    https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/304. Defined
+    in I-D.draft-ietf-ppm-dap-02. */
+    Time task_expiration;
+
+    /* A codepoint defined in I-D.draft-irtf-cfrg-vdaf-03 or reserved for private
+    use. */
+    VdafType vdaf_type;
+
+    /* Additional parameters relevant for the vdaf_type. */
+    opaque vdaf_config<1..2^16-1>;
+} TaskConfig;
+
+struct {
+    QueryType query_type;    // Defined in I-D.draft-ietf-ppm-dap-02
+    Duration time_precision; // Defined in I-D.draft-ietf-ppm-dap-02
+    uint32 min_batch_size;
+    select (query_type) {
+        case time-interval: Empty;
+        case fixed-size: uint32 max_batch_size;
+    }
+} QueryConfig;
 ~~~
 
+The purpose of `TaskConfig` is to define all parameters that are necessary
+for configuring an aggregator. It includes all the fields to be
+associated with a task (see task configuration in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.). Besides, `TaskConfig` also includes fields
+that useful for configuring a task in-band:
+
+* An opaque `task_info` that is specific to a task. For e.g. this can be a
+string describing the purpose of this task. It can also be the DAP task ID
+chosen by task author.
+
+* An opaque `vdaf_config` that contains any VDAF specific parameters for the
+chosen `vdaf_type`. The aggregators MUST pass `vdaf_config` to VDAF initialiser,
+based on the chosen `vdaf_type`.
+
+The codepoints for standardized (V)DAFs are listed below:
+
+~~~
+/* Codepoint for each standardized VDAF. Defined in
+ I-D.draft-irtf-cfrg-vdaf-02 */
+enum {
+    Prio3Aes128Count(0x00000000),
+    Prio3Aes128Sum(0x00000001),
+    Prio3Aes128Histogram(0x00000002),
+    Poplar1Aes128(0x00001000),
+    (2^32-1)
+} VdafType;
+~~~
+
+`VdafConfig` is not specified in this document, instead it should be defined by
+each VDAF implementation, for example, the simplest VDAF config for Prio3 can
+be defined as:
+
+~~~
+struct {
+    select (vdaf_type) {
+        case Prio3Aes128Count: Empty;
+        case Prio3Aes128Sum: uint8 bits;
+        case Prio3Aes128Histogram: uint32 buckets;
+    }
+} VdafConfig;
+~~~
+
+> OPEN ISSUE: Should DP parameters be defined as a different "dimension" to
+> VDAF, given that it's likely various DP mechanisms can be applied to any VDAF.
```

Comment:
That probably makes since, but it somewhat depends on the outcome of https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94. Let's keep this issue open and add a reference to the VDAF issue.